activestorage/app/javascript/activestorage/ujs.js

```diff
@@ -49,7 +49,8 @@ function handleFormSubmissionEvent(event) {
 }
 
 function submitForm(form) {
-  let button = findElement(form, "input[type=submit]")
+  let { value, name } = form._ujsData["ujs:submit-button"]
```

Comment:
`activestorage.js` doesn't depend on rails-ujs so we can't rely on help from it. Even if we could, those "private" expando properties aren't meant to be accessed directly.

Maybe there's an easier way to track which submit button was clicked? Something like:
```js
let button = findElement(form, "input[type=submit]:active, input[type=submit]")
```
(untested!)

Reply:
Yes, thats understandable.